### PR TITLE
Refactoring error handling for API requests

### LINKED ISSUE #252: No error displayed when API doesn't allow request
This can be tested by disabling one ore more addConfigurationPart lines in ApiAuthorizationFilter.java like the following line:

```
		addConfigurationPart("GET/"  + ApiServlet.LADYBUG_API_PATH + "/testtool.*$", observerRoles);
```

Enable the following line in simplelogger.properties to have better overview on what is not allowed in the Tomcat window:

```
org.slf4j.simpleLogger.log.org.apache.cxf.interceptor=WARN
```

Please note that the ApiAuthorizationFilter cannot return another return code than 200 because of a limitation in CXF (see comment below Return unauthorized" in ApiAuthorizationFilter
- [ ] This also happens when something is wrong in Quarkus

### PR DESCRIPTION
**Pull request by MarkvdVorst**
Closes #252 
Moved error handling of API calls to the http service. It is now done with pipes as this manages to catch the 401 errors, while the previous code that relied on the "error" method within the subscribe method could not detect the 401 error.
Toasts will now be shown for 401 errors.

Moved some tests to debug folder as they would only be working on that page and changed the naming of the tests to follow conventions.

### CODE REVIEW

File: src/app/test/test.component.ts [hunk 1]
```diff
@@ -155,11 +152,6 @@ export class TestComponent implements OnInit {
         next: (response: TestResult): void => {
           report.reranReport = this.createReranReport(response);
         },
-        error: () =>
-          catchError((error: HttpErrorResponse): Observable<any> => {
-            report.error = error.message;
-            return of(error);
-          }),
```
Comment (ricklue) [inline src/app/test/test.component.ts:162]: This code can't be replaced for the handleError method, so can  you keep this code here?

File: src/app/test/test.component.ts
Comment (ricklue) [inline src/app/test/test.component.ts:1]: You forgot to remove a handleError method call in the loadData method in this component

**Review by ricklue — CHANGES_REQUESTED** [review]

File: src/app/test/test.component.ts (hunk 1 above)
Comment (MarkvdVorst) [reply src/app/test/test.component.ts]: I added this back

File: src/app/test/test.component.ts
Comment (MarkvdVorst) [reply src/app/test/test.component.ts:1]: I removed it

**Review by MatthijsSmets — COMMENTED** [review]
I thought we talked about throwing an error if we don't get a 200 status code, so it can be handled in the error part of the subscribe like it was before. Angular university has a topic in their rxjs guide that talks about this called [The Catch and Rethrow Strategy](https://blog.angular-university.io/rxjs-error-handling/)

**Review by ricklue — APPROVED** [review]

File: src/app/debug/table/table.component.ts [hunk 2]
```diff
@@ -242,49 +244,60 @@ export class TableComponent implements OnInit, OnDestroy {
...
-            this.hasTimedOut = true;
-            hasChanged = true;
-          }
-        },
-        error: () => catchError(this.errorHandler.handleError()),
-      });
+      this.httpService
+        .getReportInProgress(index)
+        .pipe(
+          tap(() => this.toastService.showSuccess(`Opened report in progress with index [${index}]`)),
```
Comment (MatthijsSmets) [inline src/app/debug/table/table.component.ts:289]: This was removed in a recent PR

File: src/app/debug/table/table.component.ts [hunk 3]
```diff
@@ -467,33 +490,47 @@ export class TableComponent implements OnInit, OnDestroy {
...
-        for (let report of data) {
-          this.openReportEvent.next(report);
-        }
-      },
-      error: () => catchError(this.errorHandler.handleError()),
-    });
+    this.httpService
+      .getLatestReports(amount, this.currentView.storageName)
+      .pipe(
+        tap(() => this.toastService.showSuccess(`Latest ${amount} reports opened!`)),
```
Comment (MatthijsSmets) [inline src/app/debug/table/table.component.ts:496]: Move this to the subscribe

File: src/app/debug/table/table.component.ts [hunk 4]
```diff
@@ -467,33 +490,47 @@ export class TableComponent implements OnInit, OnDestroy {
...
-      next: (report: Report) => {
-        this.toastService.showSuccess(`Opened report in progress with index [${index}]`);
-        this.openReportEvent.next(report);
-      },
-      error: () => catchError(this.errorHandler.handleError()),
-    });
+    this.httpService
+      .getReportInProgress(index)
+      .pipe(
+        tap(() => this.toastService.showSuccess(`Opened report in progress with index [${index}]`)),
```
Comment (MatthijsSmets) [inline src/app/debug/table/table.component.ts:512]: Move this to the subscribe

File: src/app/debug/table/table.component.ts [hunk 5]
```diff
@@ -467,33 +490,47 @@ export class TableComponent implements OnInit, OnDestroy {
...
-    this.httpService.deleteReportInProgress(index).subscribe({
-      error: () => catchError(this.errorHandler.handleError()),
-      complete: () => {
-        this.loadReportInProgressSettings();
-      },
-    });
+    this.httpService
+      .deleteReportInProgress(index)
+      .pipe(
+        tap(() => this.toastService.showSuccess(`Deleted report in progress with index [${index}]`)),
```
Comment (MatthijsSmets) [inline src/app/debug/table/table.component.ts:526]: Move this to the subscribe

File: src/app/debug/table/table.component.ts [hunk 6]
```diff
@@ -467,33 +490,47 @@ export class TableComponent implements OnInit, OnDestroy {
...
+        catchError(this.errorHandler.handleError()),
+      )
+      .subscribe({
+        complete: () => {
```
Comment (MatthijsSmets) [inline src/app/debug/table/table.component.ts:530]:
```suggestion
        next: () => {
```

File: src/app/debug/table/table.component.ts [hunk 7]
```diff
@@ -521,18 +558,23 @@ export class TableComponent implements OnInit, OnDestroy {
...
-          };
-          this.tabService.openNewTab(reportData);
-        }
-      },
-      error: () => catchError(this.errorHandler.handleError()),
-    });
+    this.httpService
+      .uploadReport(formData)
+      .pipe(
+        tap(() => this.toastService.showSuccess('Report uploaded!')),
```
Comment (MatthijsSmets) [inline src/app/debug/table/table.component.ts:564]: Move this to the subscribe

File: src/app/test/clone-modal/clone-modal.component.ts [hunk 8]
```diff
@@ -30,27 +31,35 @@ export class CloneModalComponent {
...
       message: this.variableForm.value.message,
     };
-    this.httpService.cloneReport(this.currentView.storageName, this.report.storageId, map).subscribe({
-      next: () => this.cloneReportEvent.emit(),
-      error: () => catchError(this.errorHandler.handleError()),
-    });
+    this.httpService
+      .cloneReport(this.currentView.storageName, this.report.storageId, map)
+      .pipe(
+        tap(() => this.toastService.showSuccess('Report cloned!')),
```
Comment (MatthijsSmets) [inline src/app/test/clone-modal/clone-modal.component.ts:58]: Move this to the subscribe

**Review by MatthijsSmets — CHANGES_REQUESTED** [review]
A few small things

File: src/app/debug/table/table.component.ts (hunk 2 above)
Comment (MarkvdVorst) [reply src/app/debug/table/table.component.ts]: Added the change

File: src/app/debug/table/table.component.ts (hunk 3 above)
Comment (MarkvdVorst) [reply src/app/debug/table/table.component.ts]: Added the change

File: src/app/debug/table/table.component.ts (hunk 5 above)
Comment (MarkvdVorst) [reply src/app/debug/table/table.component.ts]: Added the change

File: src/app/debug/table/table.component.ts (hunk 6 above)
Comment (MarkvdVorst) [reply src/app/debug/table/table.component.ts]: Added the change

File: src/app/debug/table/table.component.ts (hunk 7 above)
Comment (MarkvdVorst) [reply src/app/debug/table/table.component.ts]: Added the change

**Review by MatthijsSmets — APPROVED** [review]

**Pull request merged**